azkaban-common/src/test/java/azkaban/user/FileWatcherTest.java

```diff
@@ -73,14 +72,33 @@ public void registerAndTake() throws Exception {
 
     final WatchKey key = this.fileWatcher.take();
     final List<WatchEvent<?>> events = this.fileWatcher.pollEvents(key);
-    // depending on the OS & file system there may be at least 1 or 2 events even with 1 write()
-    assertThat(events.size()).isGreaterThanOrEqualTo(1);
 
-    for (WatchEvent<?> event : events) {
+    // Since we watch the entire parent dir, *and* run multiple tests in parallel, we catch *other*
```

Comment:
Is there no way to make this test write to and watch a different directory that no other tests use (maybe randomly generate a folder name or something) so we don't have to add logic to _ignore_ the noise created by other tests?

Reply:
Definitely support this. I didn't know other tests are touching the same dir (what are those?). This test could be isolated by using a junit `@Rule` to create a temporary dir, copy the file there, and watch the temporary dir instead of some shared dir. Like here:

https://github.com/azkaban/azkaban/blob/7efe67b0602fdc65f18fc0e2f91e14a57c2d845e/azkaban-exec-server/src/test/java/azkaban/execapp/event/LocalFlowWatcherTest.java#L31-L32

Reply:
Thanks for bringing this up. @li-ygerchikov please let me know if I can be of any use.

Reply:
All fair points. I hoped to avoid deeper changes, esp since I seem to remember @djaiswal83 citing some reasons to keep the same file/dir management as in other tests... But you both are right in saying that there must be a better way.